examples/idle.rs

```diff
@@ -0,0 +1,77 @@
+use imap::extensions::idle;
+use native_tls::TlsConnector;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "idle")]
+struct Opt {
+    // The server name to connect to
+    #[structopt(short, long)]
+    server: String,
+
+    // The port to use
+    #[structopt(short, long, default_value = "993")]
+    port: u16,
+
+    // The account username
+    #[structopt(short, long)]
+    username: String,
+
+    // The account password. In a production system passwords
+    // would normally be in a config or fetched at runtime from
+    // a password manager or user prompt and not passed on the
+    // command line.
+    #[structopt(short = "w", long)]
+    password: String,
+
+    // The mailbox to IDLE on
+    #[structopt(short, long, default_value = "INBOX")]
+    mailbox: String,
+
+    #[structopt(
+        short = "x",
+        long,
+        help = "The number of responses to receive before exiting",
+        default_value = "5"
+    )]
+    max_responses: usize,
+}
+
+fn main() {
+    let opt = Opt::from_args();
+
+    let ssl_conn = TlsConnector::builder().build().unwrap();
+    let client = imap::connect((opt.server.clone(), opt.port), opt.server, &ssl_conn)
+        .expect("Could not connect to imap server");
+    let mut imap = client
+        .login(opt.username, opt.password)
+        .expect("Could not authenticate");
+    imap.debug = true;
+    imap.select(opt.mailbox).expect("Could not select mailbox");
+
+    let idle = imap.idle().expect("Could not IDLE");
```

Comment:
How do you feel about having the callback be passed to `idle` instead of to the individual `wait_*` methods?

Reply:
I think it makes more sense to pass it to the `wait` functions? Simply because when passing a closure it seems more normal to do this:

`
let idle = imap.idle().unwrap();

let result = idle.wait_keepalive(|response| {
    // Do something
});
`

instead of this:

`
let idle = imap.idle(|response| {
    // Do something
}).unwrap();

let result = idle.wait_keepalive();
`

And also passing it to the `wait` functions means we do not have to store the callback inside the `Handle`, but this is probably not a big deal.

Do you have a preference?

Reply:
When I see it written out, no, I agree with you :)